Dockerfile

```diff
@@ -0,0 +1,37 @@
+FROM python:3.11-slim-bullseye
+
+ENV DEBIAN_FRONTEND noninteractive
+ENV TERM linux
+
+# Install distribution packages, with caching.
+# `git` is needed for `versioningit`.
+RUN rm -f /etc/apt/apt.conf.d/docker-clean; echo 'Binary::apt::APT::Keep-Downloaded-Packages "true";' > /etc/apt/apt.conf.d/keep-cache
+RUN \
+    --mount=type=cache,id=apt,sharing=locked,target=/var/cache/apt \
+    --mount=type=cache,id=apt,sharing=locked,target=/var/lib/apt \
+    true \
+    && apt-get update \
+    && apt-get install --no-install-recommends --no-install-suggests --yes git ffmpeg firefox-esr xauth xvfb
+
+# Copy sources
+COPY . /src
+
+# Install package, with caching of dependency packages.
+RUN \
+    --mount=type=cache,id=pip,target=/root/.cache/pip \
+    true \
+    && pip install --use-pep517 --prefer-binary '/src'
+
+# Uninstall Git again.
+RUN apt-get --yes remove --purge git && apt-get --yes autoremove
```

Comment:
Each command creates a layer in the docker image that contains the changes to the previous layer.

All files are still left in the previous layer and space is saved in the image.

Reply:
Good point!

Reply:
Also git is not needed for pip, or at all?

Reply:
Oh I see because `setup.py develop` was only installing dependencies.

Now pip install is copying all python files from /src to elsewhere so it now needs to run and will install dependencies when any .py/application file is changed.

Reply:
> Also git is not needed for pip, or at all?

`git` is needed when using `versioningit`, which we may bring in soon, when modernizing further.

> Oh I see because `setup.py develop` was only installing dependencies.

`setup.py develop` is deprecated. It is replaced by `pip install --editable=.`.

> Now pip install is copying all python files from /src to elsewhere so it now needs to run and will install dependencies when any .py/application file is changed.

Thanks, I've added your assessment to GH-25.